File: 2706/buy-two-chocolates.py
# 要買最便宜的2個巧克力，看能剩多少錢。如果不夠錢，就不要買
# 其實就是在迴圈裡，找最小的2個數，便知答案
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        ans1, ans2 = inf, inf # ans1是最小值，ans2是次小值
        for price in prices:
            if price <= ans1: # 比最小值更小（含相等）便能更新遞補
                ans2 = ans1 # 第1名的值變第2名
                ans1 = price # 更新第1名
            elif price < ans2: # 比第2名還小
                ans2 = price # 更新第2名
        # 巡完後，ans1是最小值，ans2是次小值
        if ans1+ans2 <= money: # 如果夠買，就買
            return money-ans1-ans2 # 回傳剩下的錢
        else: # 不夠買的話
            return money # 就不要買，剩全部的錢

File: 2706/buy-two-chocolates.c
//在要找最便宜的2個巧克力, 所以在迴圈裡,找最小的2個數,再回傳「剩多少錢」
//錢不夠買時, 就不要買, 錢都留下來。
int buyChoco(int* prices, int pricesSize, int money){
    int min1 = 200, min2 = 200; //因 prices[i]<=100, 設200當成預設值
    for(int i=0; i<pricesSize; i++){
        if(prices[i]<min1){
            min2 = min1;
            min1 = prices[i];
        }else if(prices[i]<min2){
            min2 = prices[i];
        }
    }
    if(min1+min2 > money) return money; //錢不夠花,不要買
    else return money-min1-min2; //錢夠花, return 買剩的錢
}
